kstreamplify-core/src/test/java/com/michelin/kstreamplify/error/ProcessingResultTest.java

```diff
@@ -56,6 +61,65 @@ void shouldCreateWrappedProcessingResult() {
         assertEquals(message.timestamp(), wrappedRecord.timestamp());
     }
 
+    @Test
+    void shouldCreateWrappedProcessingResultWithHeader() {
+        String value = "Value";
+        String headerKey = "MSG_HEADER";
+        String headerValue = "Header value";
+        long timestamp = System.currentTimeMillis();
+
+        Headers headers = new RecordHeaders(Collections.singletonList(
+                new RecordHeader(headerKey, headerValue.getBytes(StandardCharsets.UTF_8))));
+        
+        Record<String, String> message = new Record<>("key", value, timestamp, headers);
+        Record<String, ProcessingResult<String, Integer>> wrappedRecord = 
+                                                            ProcessingResult.wrapRecordSuccessWithHeader(message);
+        // check key
+        assertEquals(message.key(), wrappedRecord.key());
+        // check header
+        assertEquals(1, wrappedRecord.headers().toArray().length);
+        assertEquals(
+                new String(headerValue.getBytes(StandardCharsets.UTF_8), StandardCharsets.UTF_8),
+                new String(wrappedRecord.headers().lastHeader(headerKey).value(), StandardCharsets.UTF_8)
+        );
+        // Check value
+        assertNotNull(wrappedRecord.value());
+        assertTrue(wrappedRecord.value().isValid());
+        assertEquals(value, wrappedRecord.value().getValue());
+        assertNull(wrappedRecord.value().getError());
+        // Check timestamp
+        assertEquals(message.timestamp(), wrappedRecord.timestamp());
+    }
+    
+    @Test
+    void shouldCreateWrappedProcessingResultWithHeaderV2() {
```

Comment:
`shouldWrapRecordSuccessWithHeadersFromParameters` would be more convenient and compliant with the previous test